src/main/java/com/dynatrace/openkit/core/communication/BeaconSendingFlushSessionsState.java

```diff
@@ -56,10 +58,14 @@ void doExecute(BeaconSendingContext context) {
         }
 
         // flush already finished (and previously ended) sessions
+        boolean tooManyRequestsReceived = false;
         List<SessionWrapper> finishedSessions = context.getAllFinishedAndConfiguredSessions();
         for (SessionWrapper finishedSession : finishedSessions) {
-            if (finishedSession.isDataSendingAllowed()) {
-                finishedSession.sendBeacon(context.getHTTPClientProvider());
+            if (!tooManyRequestsReceived && finishedSession.isDataSendingAllowed()) {
+                StatusResponse response = finishedSession.sendBeacon(context.getHTTPClientProvider());
+                if (response.getResponseCode() == Response.HTTP_TOO_MANY_REQUESTS) {

```

Comment:
is this the same as BeaconSendingRequestUtil.isTooManyRequestsResponse(statusResponse) in  which case I would prefer the latter